### ISSUE TITLE
GitLab support

### ISSUE THREAD

**Issue by k0nserv** [wojteklu/danger-mention#3]
Hi @wojteklu,

now that Danger supports GitLab I'd be interested in your take on supporting GitLab. I am happy to do the work if you think it can be include in the plugin in a nice way.

**Comment by wojteklu** [comment]
It would be awesome. I'm not using gitlab, but it should be possible if gitlab offers html with blame information (github: https://github.com/wojteklu/danger-mention/blame/master/lib/version.rb). Then we will use regex to analyze the outputted html.

**Comment by k0nserv** [comment]
Yeah should be possible with any problems :)

**Comment by k0nserv** [comment]
Quick question. Where does `branch_for_base` come from here https://github.com/k0nserv/danger-mention/blob/master/lib/danger_plugin.rb#L64? I can't find it, is it provided by Danger?

**Comment by k0nserv** [comment]
Found the answer to my own question :)

**Comment by jk** [comment]
Just to be explicit: `branch_for_base` is only available when you're building with github. GitLab uses `branch_for_merge`.

**Comment by k0nserv** [comment]
That seems like a bug? Should they not be the same? cc @orta

**Comment by orta** [comment]
Same would be better, currently gitlab has less `branch_for_x` and different named ones ( I think I named them after the git terminology, so it should be fine to apply to both )

http://danger.systems/reference.html

**Comment by k0nserv** [comment]
Yeah that's my bad, I think I based my GitLab work on the GitHub plugin, but maybe I didn't. I'll make a backwards compatible PR that introduces the same names

**Comment by orta** [comment]
Rockin' 👍

**Comment by connorshea** [comment]
Once Danger supports GitLab CI (or rather, GitLab CI supports Danger 😉) we'd love to be able to use this plugin for GitLab CE.

If there's anything we can do to help we'd be more than willing :)

**Comment by orta** [comment]
This is added in 0.6.0 and shipped just now

![screen shot 2017-02-10 at 3 28 51 pm](https://cloud.githubusercontent.com/assets/49038/22843028/a79be5ea-efa5-11e6-9816-3a24370344fc.png)

**Comment by orta** [comment]
I take this back, as it uses web-page scraping #11 , I don't think the changes I've added are enough for gitlab support
